dapplet/src/index.ts

```diff
@@ -151,121 +145,214 @@ export default class TwitterFeature {
+        me.hidden = true;
+      } else {
+        me.label = 'Link';
+        me.tooltip = `Link ${
+          parsedNearAccount ? parsedNearAccount + ' ' : ''
+        }account to @${username} using Connected Accounts smart contract`;
+        me.hidden = false;
       }
-      me.hidden = false;
     } else {
       me.hidden = true;
     }
   };
 
-  onProfileButtonLinkExec = async (profile, me) => {
-    const nearAccount = this.parseNearId(profile.authorFullname, this._network);
+  onProfileButtonLinkExec = async (profile: { id: string; authorFullname: string }, me) => {
+    const parsiedNearAccount = this.parseNearId(profile.authorFullname, this._network);
+    const { websiteName, fullname, img } = await this.getCurrentUserAsync();
+    const websiteNameLowerCase = websiteName.toLowerCase();
     try {
+      const accountGId = profile.id + '/' + websiteName.toLowerCase();
+      const walletForAutoclaim = await this.tippingService.getWalletForAutoclaim(accountGId);
       me.disabled = true;
       me.loading = true;
-      const linkedAccount = await this.identityService.getNearAccount('twitter/' + profile.id);
       me.label = 'Waiting...';
-      if (linkedAccount) {
+      if (walletForAutoclaim) {
         // unlink
-        if (nearAccount) {
-          alert('Remove NEAR Account ID from your profile name before continue.');
-        } else {
-          await this.identityService.requestVerification(
-            `twitter/${profile.id}`,
-            true,
-            'https://twitter.com/' + profile.id,
-          );
-        }
+        await this.tippingService.deleteWalletForAutoclaim(accountGId);
       } else {
         // link
-        if (!nearAccount) {
-          const exampleWallet = this._network === NearNetwork.TESTNET ? 'yourwallet.testnet' : 'yourwallet.near';
+        if (!parsiedNearAccount) {
+          const exampleWallet = this._network === NearNetworks.Testnet ? 'yourwallet.testnet' : 'yourwallet.near';
           alert(
-            'Add your NEAR account ID to your profile name in Twitter before continuing. ' +
+            `Before you continue, add your NEAR account ID to your ${websiteName} profile name. ` +
               'This is necessary for Oracle so that it can make sure that you own this Twitter account. ' +
               'After linking you can remove it back.\n' +
               `For example: "${profile.authorFullname} (${exampleWallet})"\n`,
           );
         } else {
-          await this.identityService.requestVerification(
-            `twitter/${profile.id}`,
-            false,
-            'https://twitter.com/' + profile.id,
-          );
+          const prevSessions = await Core.sessions();
+          const prevSession = prevSessions.find((x) => x.authMethod === 'near/' + this._network);
+          const session = prevSession ?? (await Core.login({ authMethods: ['near/' + this._network] }));
+          const wallet = await session.wallet();
```

Comment:
Can this logic be moved to a separate service to make the dapplet code cleaner?

Reply:
fixed